src/components/views/rooms/SearchResultTile.js

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2015 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+'use strict';
+
+var React = require('react');
+var sdk = require('../../../index');
+
+module.exports = React.createClass({
+    displayName: 'SearchResult',
+
+    propTypes: {
+        searchResult: React.PropTypes.object.isRequired,
+        searchHighlights: React.PropTypes.array,
+        onClick: React.PropTypes.func,
```

Comment:
right now, the thing being clicked on is the highlight. But that is an implementation detail of how the search result is presented. There are other ways that a user might plausibly choose this search result from the list, and I want to keep that decision inside SearchResultTile, rather than split between SearchResultTile and RoomView.